internal/schema/schema.go

```diff
@@ -76,18 +84,39 @@ func NewSchemaMigrationService(
 	roleService RoleService,
 	actionService ActionService,
 	policyService PolicyService,
-	authzEngine AuthzEngine) *SchemaService {
+	authzEngine AuthzEngine,
+	userRepository UserRepository,
+	defaultSystemEmail string) *SchemaService {
```

Comment:
lets pass the app Config itself rather than the string

Reply:
this tends to lead to import circle but let me see if I can do something

Reply:
tends to cause dependency cycle but will try

Reply:
Oh right, if that is the case then we could define a config for `SchemaMigrationConfig` and put this config during the construction. Basically doing inverse dependency.